src/pyOpenMS/CMakeLists.txt

```diff
@@ -678,6 +678,10 @@ else()
         LIBRARY_OUTPUT_DIRECTORY_DEBUG "${CMAKE_BINARY_DIR}/pyOpenMS/pyopenms"
         RUNTIME_OUTPUT_DIRECTORY_DEBUG "${CMAKE_BINARY_DIR}/pyOpenMS/pyopenms"
     )
+    add_custom_command(TARGET pyopenms_compile
```

Comment:
The idea is, that this is not needed since everything is in one module.

Reply:
I cannot judge since I never tested but in theory python.typing should find all classes.

Reply:
I'm out of my depth in how the resolution is supposed to work. But adding that seems to have solved the compilation problem, and changing the create_cpp_extension seems to fix the naming issue.

Reply:
That is super strange. The pyi files should not have anything to do with compilation.
I guess it won't hurt.
Can you add a Todo: check if really necessary.?

Reply:
Done.